### ISSUE TITLE
How to detect when image uploaded?

### ISSUE THREAD

**Issue by sebastialonso** [adonespitogo/angular-base64-upload#20]
First of all, great module! Neat and simple!

I tried using `$watch`, but eventually I get

~~~
Error: [$rootScope:infdig] 10 $digest() iterations reached. Aborting!
~~~
 Is there an efficient way to detect when an image was *uploaded* to the uploader?

**Comment by adonespitogo** [comment]
I think you could do it simply with the $http success callback since you are sending a basic post request to the server:

```
        $http.post('server.php', $scope.base64Object)
        .success(function(res){
          alert('File successfully uploaded!');
        })
```

**Comment by sebastialonso** [comment]
Thanks for your answer. I'm afraid I completely ruined what I was originally asking.

I meant if there was an easy way to detect when the user changes the image they had previously assigned to the file input.

Something like:
* User puts picture
* User doesn't like picture, decides for another one
* User substitutes former picture for a new one
* A random event is fired
* User uploads data to the server

Either way, I solved it manually, by $watching an attribute of the object, say filesize.

**Comment by adonespitogo** [comment]
Good catch. Sorry there's no event fired when a user changes the chosen file currently. What you did is the best way I can think of to solve your problem as well. I'll mark this issue as an enhancement for now.

**Comment by sebastialonso** [comment]
Sure, don't worry about it. I love the simplicity of your module. I'll try to read the code on my spare time to see if I can help.

Cheers!

**Comment by adonespitogo** [comment]
Thanks man. Cheers!

**Comment by adonespitogo** [comment]
#27 There are now many ways to get around this, either to listen to file reader events or watch the models.

**Comment by adonespitogo** [comment]
Fixed in pull request #28 (version 0.1.1)

```
$scope.onChangeHandler = function (e, fileList) {
  console.log(e);
  console.log(fileList);
};

<input base-sixty-four-input ng-model="file" on-change="onChangeHandler">
```
